spring-cloud-kubernetes-client-config/src/test/java/org/springframework/cloud/kubernetes/client/config/reload/KubernetesClientEventBasedConfigMapChangeDetectorTests.java

```diff
@@ -142,7 +143,9 @@ void watch() throws Exception {
 				mock(KubernetesClientConfigMapPropertySource.class)).withProperty("debug", "true");
 		KubernetesClientConfigMapPropertySourceLocator locator = mock(
 				KubernetesClientConfigMapPropertySourceLocator.class);
-		when(locator.locate(environment)).thenReturn(new MockPropertySource().withProperty("debug", "false"));
```

Comment:
change needed because the raw type in the caller has been removed, as such this would not compile; so use `thenAnswer` instead of `thenReturn`. The test logic does not change, but now we can compile.